wordpress/wp-content/plugins/cds-wpml-mods/resources/js/components/PageSelect.js

```diff
@@ -0,0 +1,159 @@
+const { SelectControl } = wp.components;
+const { useState, useEffect } = wp.element;
+const { select, subscribe } = wp.data;
+const { __ } = wp.i18n;
+
+import { getData, sendData } from '../util/fetch.js';
+
+export const PageSelect = () => {
+    const { isSavingPost } = select( 'core/editor' );
+
+    const emptyPage = { is_translated: null, value: 0, label: __("None", "cds-wpml-mods") };
+    // get the current post id
+    const postID = select("core/editor").getCurrentPostId();
+    const type = select("core/editor").getCurrentPostType();
+
+    const hintTexts = {
+        'empty': () => `${__('No translation assigned for this', "cds-wpml-mods")} ${type}.`,
+        'untranslated': (label) => `“${label}” ${__('will be assigned as translation for this', "cds-wpml-mods")} ${type}.`,
+        'translated': (label) => `${__('⚠️ This will unlink the existing translation for', "cds-wpml-mods")} “${label}”.`,
+        'translated_post_id': (label) => `“${label}” ${__('is the current translation for this', "cds-wpml-mods")} ${type}.`,
+    }
+
+    const [isLoading, setIsLoading] = useState(true);
+    const [post, setPost] = useState();
+    const [pages, setPages] = useState([emptyPage]);
+    const [page, setPage] = useState(emptyPage);
+    const [hintText, setHintText] = useState(hintTexts['empty']());
+    const [isSavingProcess, setSavingProcess] = useState(false);
+
+    /* Triggers when the "publish" or "update" button is clicked */
+    /* https://github.com/WordPress/gutenberg/issues/17632#issuecomment-1153888435 */
+    subscribe(() => {
+        if (isSavingPost()) {
+            setSavingProcess(true);
+        } else {
+            setSavingProcess(false);
+        }
+    });
+
+    const updatePost = async () => {
+        setIsLoading(true)
+        const response = await sendData(`cds/wpml/posts/${postID}/translation`, {translationId: page.value})
```

Comment:
Looks like there's still an issue here when setting back to `None` ya?
<img width="500" alt="Screen Shot 2022-07-29 at 10 09 14 AM" src="https://user-images.githubusercontent.com/62242/181778540-2527d7c1-d06a-4267-aa04-703b8d483785.png">

Reply:
Yeah, there's no API route for it yet. We can ship it without this working assuming nobody needs to do this for a day or two.